### PR TITLE
add api client and model for cost cred

### LINKED ISSUE #218: support cloud cost credentials
We'd like to add support for our cost credentials.
We'd need the following new resources (and data for each resource)

### cost_credentials_project_assignment

Exactly the same as `cloud_credentials_project_assignment` resource and data. The only difference is that we use different APIs:
* [Create](https://developer.env0.com/docs/api/b3A6MTkyNjc1NzI-associate-cost-credentials-with-project)
* [Delete](https://developer.env0.com/docs/api/b3A6MTkyNjc1NzU-dissociate-cost-credentials-from-project)
* [Read](https://developer.env0.com/docs/api/b3A6MTkyNjc1NzM-list-cost-credentials-associated-to-project) - Notice the response format is different than [this](https://developer.env0.com/docs/api/b3A6MTkyNjc1NzE-list-deployment-a-project-s-deployment-credentials)

### aws_cost_credentials

Exactly the same as `aws_credentials` resource and data (which is a deploy credential, not a cost credential). The schema is exactly the same, and the API calls are the same. The only difference is that the `type` that should be `AWS_ASSUMED_ROLE`:
* [Create](https://developer.env0.com/docs/api/b3A6ODc4NjAwOQ-create-cloud-provider-cost-credentials) should be called with correct `type` and data in `value`
* data should filter by type = `AWS_ASSUMED_ROLE`

### azure_cost_credentials

Similar to `aws_cost_credentials` resource and data. Should use the same APIs, where type is `AZURE_CREDENTIALS `. For the schema - see [docs](https://developer.env0.com/docs/api/b3A6ODc4NjAwOQ-create-cloud-provider-cost-credentials) - value:
* clientId - required
* clientSecret - required and sensitive
* tenantId - required
* subscriptionId - required
Other than the schema, differences in API calls are as follows:
* [Create](https://developer.env0.com/docs/api/b3A6ODc4NjAwOQ-create-cloud-provider-cost-credentials) should be called with correct `type` and data in `value`
* data should filter by type = `AZURE_CREDENTIALS `

### google_cost_credentials

Similar to `aws_cost_credentials` resource and data. Should use the same APIs, where type is `GCP_CREDENTIALS`. For the schema - see [docs](https://developer.env0.com/docs/api/b3A6ODc4NjAwOQ-create-cloud-provider-cost-credentials) - value:
* tableId - required
* secret - required and sensitive
Other than the schema, differences in API calls are as follows:
* [Create](https://developer.env0.com/docs/api/b3A6ODc4NjAwOQ-create-cloud-provider-cost-credentials) should be called with correct `type` and data in `value`
* data should filter by type = `GCP_CREDENTIALS`

### PR DESCRIPTION
**Pull request by samuel-br**
### Issue & Steps to Reproduce / Feature Request
[//]: <> (choose “resolves” “fixes” or “closes” and put link for the issue)
This PR is part of other PRs when  combined fix #218 
fixes #218 
### Solution
This PR is part of the solution for adding cost cred resources and cost assignment to project

1. - add to model structs supporting those resources 
2. - add API call to `cloud_credentials ` for google cost credentials
3. - add file `cost_credentials_project_assignment` to client for APIs calls
4. - add to `api-client` the relevant function mentioned in 2,3

### CODE REVIEW

**Review by eranelbaz — COMMENTED** [review]
update the description as this PR doesn't fix the issue,
its part of multiple PRs that will

File: client/api_client.go [hunk 1]
```diff
@@ -41,10 +41,14 @@ type ApiClientInterface interface {
 	AwsCredentialsCreate(request AwsCredentialsCreatePayload) (ApiKey, error)
 	CloudCredentialsDelete(id string) error
 	GcpCredentialsCreate(request GcpCredentialsCreatePayload) (ApiKey, error)
+	GoogleCostCredentialsCreate(request GoogleCostCredentialsCreatePayload) (ApiKey, error)
```
[the remark this answers is not in the document]
Comment (samuel-br) [reply client/api_client.go:44]: not needed cause `AzureCredentialsCreate ` and `AwsCredentialsCreate` do the  same job

File: client/model.go [hunk 2]
```diff
@@ -249,6 +249,12 @@ type CloudCredentialsProjectAssignment struct {
 	ProjectId    string `json:"projectId"`
 }
 
+type CostCredentialProjectAssignment struct {
+	ProjectId       string `json:"projectId"`
+	CredentialsId   string `json:"credentialsId"`
+	CredentialsType string `json:"credentialsType"`
+}
```
[the remark this answers is not in the document]
Comment (samuel-br) [reply client/model.go:256]: I write it there so it will be next to other cred assignment. WDYT

**Review by eranelbaz — COMMENTED** [review]
Please add tests

**Comment by samuel-br** [comment]
> Please add tests

4996a31fc4a9df8d3c07ad0dd99ea262c0dce4ba

**Review by eranelbaz — APPROVED** [review]

**Pull request merged**